src/View/ViewBuilder.php

```diff
@@ -301,25 +302,58 @@ public function getPlugin(): ?string
     }
 
     /**
-     * Adds a helper to use.
+     * Adds a helper to use if no such helper with this name already exists.
      *
      * @param string $helper Helper to use.
      * @param array<string, mixed> $options Options.
      * @return $this
+     * @throws \Cake\Core\Exception\CakeException When a duplicate is found.
      * @since 4.1.0
      */
     public function addHelper(string $helper, array $options = [])
     {
-        $this->_helpers[$helper] = $options;
+        [$plugin, $name] = pluginSplit($helper);
+        if (isset($this->_helpers[$name])) {
+            $message = sprintf('The `%s` alias has already been added as helper. Use `setHelper()` instead.', $name);
+
+            throw new CakeException($message);
+        }
+
+        if ($plugin) {
+            $options['className'] = $helper;
+        }
+
+        $this->_helpers[$name] = $options;
+
+        return $this;
+    }
+
+    /**
+     * Adds a helper to use, overwriting any existing one with that name.
+     *
+     * @param string $helper Helper to use.
+     * @param array<string, mixed> $options Options.
+     * @return $this
+     * @since 4.4.0
+     */
+    public function setHelper(string $helper, array $options = [])
```

Comment:
Do we need both methods? Some other approaches that come to mind:

1. Start overwriting helper configuration data in `addHelper()` and only have one method instead of two. This is much harder to have deprecations around as the merge/overwrite intent is impossible to determine.
2. Add `setHelper()` and deprecate `addHelper()` in 4.4. This is a noisy change to make as almost every application will require updates and for *most* use cases it won't matter as I think few people are actually relying on this unfortunate mangling of configuration data.

Personally I prefer fewer methods especially because `addHelper()` is gaining a method order restriction that also creates upgrade friction as it adds an exception to the same rare scenario that we would modify behavior in a breaking way.

Reply:
So, adding addHelper() with a bool argument to "overwrite" instead?

I think, given the clear API we have in other places, add vs set here seems like a fair addition to it. Quite expected and a bit simpler to use than the boolean flag to overwrite.

Of course, we wouldnt need to if we allowed add to silently replace. But from I heard we would like to have this by default reported for easier debugging of issues.

Reply:
> Of course, we wouldnt need to if we allowed add to silently replace. But from I heard we would like to have this by default reported for easier debugging of issues.

I am ok with silent replacement as I think that should have been the behavior all along. I also believe that few applications would be impacted by the configuration being overwritten as I don't see it as a feature we have included in the documentation.

Reply:
I don't have a strong opinion in favour of keeping the error on duplicate. So I too am fine with the silent overwrite.